x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -466,25 +480,33 @@ void validateApiKeyCredentials(String docId, Map<String, Object> source, ApiKeyC
                                 validateApiKeyCredentials(docId, source, credentials, clock, listener);
                             }
                         }, listener::onFailure),
-                        threadPool.generic(), threadPool.getThreadContext());
+                        threadPool.executor(THREAD_POOL_NAME), threadPool.getThreadContext());
```

Comment:
The specified thread pool will only be used when the future is being computed, i.e. expensive hashing is underway. In this case, the API key is not yet cached, so change it to use the separate hashing pool which are dedicated to uncached API keys.

Reply:
Reverted it back to use generic pool. I don't know exactly the best practice of using generic pool. I think what you said makes sense. But on the other hand, generic pool being unbounded is likely an issue by itself. It is likely a bad circuit breaker. Anyway, we have been using generic pool here for a while now, so it is safe to keep it untouched.